app/services/message_templates/hook_execution_service.rb

```diff
@@ -71,6 +71,9 @@ def should_send_csat_survey?
     # only send CSAT once in a conversation
     return if conversation.messages.where(content_type: :input_csat).present?
 
+    # only send CSAT if customer can still reply
```

Comment:
Can we add an activity message so that the agent is clear that the CSAT is not sent due to 24 hour restriction?

Reply:
I'm pretty sure that there will be support requests in the future to figure out why we are not sending CSAT messages.

Reply:
Makes sense. Made the changes.